CMakeLists.txt

```diff
@@ -0,0 +1,45 @@
+# This source file is part of the SwiftCrypto open source project
+#
+# Copyright (c) 2021 Apple Inc. and the SwiftCrypto project authors
+# Licensed under Apache License v2.0
+#
+# See LICENSE.txt for license information
+# See CONTRIBUTORS.md for the list of SwiftCrypto project authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.15.1)
+
+project(SwiftCrypto
+  LANGUAGES C Swift)
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
+
+include(SwiftSupport)
+
+# Fixes error: "requires dynamic R_X86_64_PC32 reloc against 'XXX' which may overflow at runtime"
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
```

Comment:
Ah, yes, we should definitely be building in PIC mode, but this is the wrong way to accomplish that.  You should use the following:

```cmake
set(CMAKE_POSITION_INDEPENDENT_CODE YES)
```

However, that is not entirely for free either.  I think that for static builds, where SwiftCrypto is not being exported, it makes sense to be non-PIC (as it will be internal, the linker should be able to resolve everything relatively.  We could try something like this I suppose:

```cmake
if(BUILD_SHARED_LIBS)
  set(CMAKE_POSITION_INDEPENDENT_CODE YES)
endif()
```